src/base.cc

```diff
@@ -63,15 +63,7 @@ void assertOK(AstObject *rawPtr1, AstObject *rawPtr2) {
 }
 
 ConstArray2D arrayFromVector(std::vector<double> const &vec, int nAxes) {
-    if (vec.size() % nAxes != 0) {
-        std::ostringstream os;
-        os << "vec length = " << vec.size() << " not a multiple of nAxes = " << nAxes;
-        throw std::runtime_error(os.str());
-    }
-    int nPoints = vec.size() / nAxes;
-    Array2D::Index shape = ndarray::makeVector(nPoints, nAxes);
-    Array2D::Index strides = ndarray::makeVector(nAxes, 1);
-    return external(vec.data(), shape, strides);
+    return arrayFromVector(const_cast<std::vector<double> &>(vec), nAxes);
 }
```

Comment:
While this is correct and works fine, I personally would have liked to see this as a templated function call, specialized on the const and non const versions. I think it would be a bit clearer in reading, and clearer in return types (as it is right now calling the const overload will call the unconst overload, which returns an Array2D object which the compiler automatically casts to a ConstArray2D, as it is the declared return type).